include/tvm/compilation.h

```diff
@@ -0,0 +1,211 @@
+    std::unordered_set<std::string> options({
+        "-device=rasp",
+        "-mtriple=armv7l-none-linux-gnueabihf",
+        "-mcpu=cortex-a53",
+        "-mattr=+neon"
+    });
+    return Target("llvm", kDLCPU, 512, 1, keys, options);
+}
+
+/*! \return A target for stackvm */
+Target target_stackvm() {
+    std::unordered_set<std::string> keys({ "stackvm", "cpu" });
+    std::unordered_set<std::string> options;
+    return Target("stackvm", kDLCPU, 512, 1, keys, options);
+}
+
+
+bool LLVMEnabled() {
+    const runtime::PackedFunc* pf = runtime::Registry::Get("codegen.llvm_target_enabled");
+    if (pf == nullptr) return false;
+    return (*pf)("llvm");
+}
+
+/*! \return The default host target for a given device target */
+Target default_target_host(Target target) {
+    if (target.deviceType == kDLCPU) {
+        return target;
+    } else {
+        if (LLVMEnabled()) {
+            return target_llvm();
+        } else {
+            return target_stackvm();
+        }
+    }
+}
+
+
+/*!
+* \brief Container for build configuration options
+*/
+struct BuildConfig {
+    int data_alignment = -1;
+    int offset_factor = 0;
+
+    int double_buffer_split_loop = 1;
+    int auto_unroll_max_step = 0;
+    int auto_unroll_max_depth = 8;
+    int auto_unroll_max_extent = 0;
+    bool unroll_explicit = true;
+
+    bool restricted_func = true;
+
+    bool detect_global_barrier = false;
+
+    BuildConfig() {
+    }
+};
+
+
+/*! \brief Convenience function for getting attributes */
+TVMValue GetAttr(NodeRef node, std::string attrName) {
+    TVMValue attrValue;
+    int typeCode, success;
+
+    TVMNodeGetAttr((NodeHandle)&node, attrName.c_str(), &attrValue, &typeCode, &success);
+    return attrValue;
+}
+
+/*! \brief Convenience function for getting handle attributes */
+template<typename T>
+T GetAttrHandle(NodeRef node, std::string attrName) {
+    auto attrValue = GetAttr(node, attrName);
+    return *(reinterpret_cast<T*>(attrValue.v_handle));
+}
+
+
+/*!
+* \brief Build a Stmt given a schedule, args and binds. This function runs the IR passes.
+* \param sch The schedule to build.
+* \param args The arguments for the schedule.
+* \param binds Buffer assignments.
+* \param loopPartition True if the LoopPartition pass should be included.
+* \param argListOut Returns the arguments for the Stmt.
+* \param config The build configuration.
+* \return The built Stmt.
+*/
+EXPORT Stmt BuildStmt(Schedule sch, Array<Tensor> args, std::unordered_map<Tensor, Buffer> binds,
+    bool loopPartition, Array<NodeRef> *argListOut, const BuildConfig& config);
+
+/*!
+* \brief Build a LoweredFunc given a schedule, args and binds
+* \param sch The schedule to lower.
+* \param args The arguments to the function.
+* \param name The name of the lowered function.
+* \param binds Buffer assignments.
+* \param config The build configuration.
+* \return The lowered function.
+*/
+EXPORT LoweredFunc Lower(Schedule sch, Array<Tensor> args, std::string name,
```

Comment:
Lower->lower as it is user facing, make API function consistent with python API. Pass by const reference when possible if the function do not move the content

Reply:
returns Array<LoweredFunc>